src/types/redis_stream.cc

```diff
@@ -1619,11 +1672,12 @@ uint64_t Stream::trim(engine::Context &ctx, const std::string &ns_key, const Str
     metadata->recorded_first_entry_id.Clear();
   }
 
-  if (ret > 0) {
+  if (delete_cnt > 0) {
     metadata->max_deleted_entry_id = entryIDFromInternalKey(last_deleted);
   }
 
-  return ret;
+  return rocksdb::Status::OK();
+  ;
```

Comment:
I tried it and I can't use `StatusOr<T>` as the return value because `rocksdb::Status` and `Status` are different and `rocksdb::Status` lacks methods like `GetValue()`.

Reply:
I see. Can you change `&delete_cnt` to `*delete_cnt`?